include/lbann/execution_algorithms/ltfb/regularized_evolution.hpp

```diff
@@ -0,0 +1,117 @@
+////////////////////////////////////////////////////////////////////////////////
+// Copyright (c) 2014-2021, Lawrence Livermore National Security, LLC.
+// Produced at the Lawrence Livermore National Laboratory.
+// Written by the LBANN Research Team (B. Van Essen, et al.) listed in
+// the CONTRIBUTORS file. <lbann-dev@llnl.gov>
+//
+// LLNL-CODE-697807.
+// All rights reserved.
+//
+// This file is part of LBANN: Livermore Big Artificial Neural Network
+// Toolkit. For details, see http://software.llnl.gov/LBANN or
+// https://github.com/LLNL/LBANN.
+//
+// Licensed under the Apache License, Version 2.0 (the "Licensee"); you
+// may not use this file except in compliance with the License.  You may
+// obtain a copy of the License at:
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or
+// implied. See the License for the specific language governing
+// permissions and limitations under the license.
+////////////////////////////////////////////////////////////////////////////////
+#ifndef LBANN_EXECUTION_ALGORITHMS_LTFB_REGULARIZED_EVOLUTION_HPP_INCLUDED
+#define LBANN_EXECUTION_ALGORITHMS_LTFB_REGULARIZED_EVOLUTION_HPP_INCLUDED
+
+#include "mutation_strategy.hpp"
+
+#include "meta_learning_strategy.hpp"
+
+#include <google/protobuf/message.h>
+
+#include <cstddef>
+#include <memory>
+#include <string>
+#include <unordered_map>
+
+namespace lbann {
+namespace ltfb {
+
+/** @class RegularizedEvolution
+ *  INSERT DESC HERE
+ */
+
+class RegularizedEvolution final
+  : public Cloneable<RegularizedEvolution, MetaLearningStrategy>
+{
+public:
+  enum class metric_strategy
+  {
+    LOWER_IS_BETTER,
+    HIGHER_IS_BETTER,
+  }; // enum class metric_strategy
+
+public:
+  RegularizedEvolution(std::string metric_name,
+                       metric_strategy winner_strategy,
+                       std::unique_ptr<MutationStrategy> mutate_algo,
+                       int sample_size);
+
+  RegularizedEvolution(std::unordered_map<std::string, metric_strategy> metrics,
+                       std::unique_ptr<MutationStrategy> mutate_algo,
+                       int sample_size);
+
```

Comment:
Multiple metrics don't make sense in regularized evolution, so we shouldn't expose that option:
```suggestion
```

Reply:
On second thought, I could imagine multiple metrics being valid if we have a more advanced selection scheme, like picking a random model from the Pareto front. That said, we don't support this case right now so we should either remove this option or throw an exception if the algorithm receives more than one metric.

Reply:
Removed